src/lipalightninglib.udl

```diff
@@ -227,6 +227,22 @@ dictionary LspFee {
     u64 channel_fee_permyriad; // Parts per myriad (aka basis points) -> 100 is 1%
 };
 
+dictionary PaymentAmountLimits {
+    u64 max_receive_sat; // Hard limit: The maximum sat amount a user is allowed to receive per payment
+    ChannelRelatedLimit? channel_related_limit; // Only provided if inbound capacity < max_receive
+};
+
+dictionary ChannelRelatedLimit {
+    AmountLimitType limit_type;
+    u64 amount_sat;
+};
+
+enum AmountLimitType {
+    "MaxFreeReceive", // Soft limit: The maximum sat amount a user can receive without being charged a setup fee
+    "MinReceive", // Hard limit: The minimum sat amount a user must receive with the next payment
+        // If this limit is provided, that means a setup fee will be charged for the incoming payment
+};
```

Comment:
Would it make sense to have an enum for the channel-based limit?

For example:
```suggestion
dictionary PaymentAmountLimits {
    u64 max_receive_sat; // Hard limit: The maximum sat amount a user is allowed to receive per payment
    ChannelRelatedLimit channel_related_limit;
};

[Enum]
interface ChannelRelatedLimit {
    NoLimit(); // inbound capacity >= max_receive_sat
    MaxFreeReceive(u64 limit);
    MinReceive(u64 limit);
};
```

Reply:
Yes, that's probably the more idiomatic way

Reply:
Maybe `ChannelRelatedLimit` could even be `LiquidityLimit`, but I'm fine with the current name if you prefer it :)

Reply:
Yes I like `LiquidityLimit` better. Couldn't find a good term myself. Thanks!

Check out 49546d2b53d16fde7b260b9861a7c1d480f4d915